### PR TITLE
Add an example that uses napa.sync.Lock

### PR DESCRIPTION
**Pull request by fs-eire**
This is an example of using `napa.sync.Lock`. It implemented a simple lazy-loading component that can be shared across napa zones.

### CODE REVIEW

**Comment by helloshuangzi** [comment]
Overall, the implementation looks good to me.
Meanwhile, as my personal opinion, it may be better to make the example go with a common scenario instead of using "do_something" / "foo" / "bar", etc. 
You may consider to make it a "phone book look up program" to load <person, number> lazily, and use "lookUp" to repleace "do_something". This would get the users to the program more easier.

BTW, if the function name is not that too long, I would suggest to use "camelCase".

**Comment by helloshuangzi** [comment]
Add the example link to ./README.md

File: examples/tutorial/lazy-loading/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Lazy Loading
```
Comment (daiyip) [inline examples/tutorial/lazy-loading/README.md]: Better to choose a title that can capture users' mind about "synchronization".
To me,  'lazy' here is not that important, it's just a implementation detail.
how about 'synced-loading'?

File: examples/tutorial/synchronized-loading/phone-book-data.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "ashley": "123-456-7890",
+    "david": "123-444-5555",
+    "lisa": "567-888-9999",
+    "tony": "456-789-0000"
+}
```
Comment (helloshuangzi) [inline examples/tutorial/synchronized-loading/phone-book-data.json]: add am empty line at EOF

File: README.md [hunk 3]
```diff
@@ -31,9 +31,10 @@ zone1.execute(
 ```
 More examples:
 * [Estimate PI in parallel](./examples/tutorial/estimate-pi-in-parallel)
+* [Max sub-matrix of 1s with layered parallelism](./examples/tutorial/max-square-sub-matrix)
 * [Parallel Quick Sort](./examples/tutorial/parallel-quick-sort)
 * [Recursive Fibonacci with multiple JavaScript threads](./examples/tutorial/recursive-fibonacci)
-* [Max sub-matrix of 1s with layered parallelism](./examples/tutorial/max-square-sub-matrix)
+* [Synchronized Loading](./examples/tutorial/synchronized-loading)
```
Comment (daiyip) [inline README.md]: L in lower case please.

File: examples/tutorial/synchronized-loading/phone-book.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    for (let name in phoneBookData) {
+        store.set(name, phoneBookData[name]);
+    }
+    store.set('_loaded', true);
+}
+
+let loaded = false;
+const lookup = function (name) {
+    if (!loaded) {
+        const lock = store.get('_lock');
```
Comment (daiyip) [inline examples/tutorial/synchronized-loading/phone-book.js]: replace _lock with a more meaningful name.

File: examples/tutorial/synchronized-loading/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Synchronized loading
+This example implements a shared phone book component. The component will not load data until the first lookup request happens. When it starts to load data, it ensures the loading will be run only once.
```
Comment (daiyip) [inline examples/tutorial/synchronized-loading/README.md]: be run-> run

**Pull request merged**